hail/src/main/scala/is/hail/shuffler/ShuffleClient.scala

```diff
@@ -15,46 +16,101 @@ object ShuffleClient {
   private[this] val log = Logger.getLogger(getClass.getName())
 
   lazy val sslContext = {
-    val key = System.getenv("SHUFFLER_SSL_CLIENT_KEY_FILE")
-    val cert = System.getenv("SHUFFLER_SSL_CLIENT_CERT_FILE")
-    if (key == null && cert != null ||
-      cert == null && key != null) {
+    val keyStore = System.getenv("SHUFFLER_SSL_CLIENT_KEY_STORE")
+    val trustStore = System.getenv("SHUFFLER_SSL_CLIENT_TRUST_STORE")
+    if (keyStore == null && trustStore != null ||
+      trustStore == null && keyStore != null) {
       fatal("you must specify both or neither of the hail context flags: " +
-        "shuffler_ssl_client_key_file and shuffler_ssl_client_cert_file")
+        "shuffler_ssl_client_key_store and shuffler_ssl_client_trust_store")
     }
-    if (key == null) {
+    if (keyStore == null) {
       is.hail.shuffler.sslContext(
         getClass.getResourceAsStream("/non-secret-key-and-trust-stores/client-keystore.p12"),
         "hailhail",
+        "PKCS12",
         getClass.getResourceAsStream("/non-secret-key-and-trust-stores/client-truststore.p12"),
-        "hailhail"
+        "hailhail",
+        "JKS"
       )
     } else {
-      is.hail.shuffler.sslContext(key, "", cert, "")
+      is.hail.shuffler.sslContext(keyStore, "", "PKCS12", trustStore, "", "PKCS12")
+    }
+  }
+
+  def socket(): Socket = {
+    var host = System.getenv("SHUFFLER_SSL_CLIENT_HOST")
+    if (host == null) {
+      host = "localhost"
+    }
+    var portStr = System.getenv("SHUFFLER_SSL_CLIENT_PORT")
+    if (portStr == null) {
+      portStr = "8080"
     }
+    val port = java.lang.Integer.valueOf(portStr)
+    socket(host, port)
   }
 
   def socket(host: String, port: Int): Socket = {
     val s = sslContext.getSocketFactory().createSocket(host, port)
     log.info(s"connected to ${host}:${port} (socket())")
     s
   }
+
+  def codeSocket(): Code[Socket] =
+    Code.invokeScalaObject0[Socket](ShuffleClient.getClass, "socket")
+
+  def openConnection(cb: EmitCodeBuilder, typ: TShuffle): (SocketCode, InputBufferValue, OutputBufferValue, LoggerCode) = {
+    val socketLocal = cb.emb.newLocal[Socket]("shuffleClientSocket")
+    cb.append(socketLocal := codeSocket())
```

Comment:
I think you can just write:

```
val socket = cb.newLocal("shuffleClientSocket", codeSocket())
```

This just declares a local with an initializer (at the end of the code builder).

This pattern shows up in a few places.